### PR TITLE
bug fix in search API: do not return duplicate projects when user does not login

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: src/common/dao/project.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func SearchProjects(userID int) ([]models.Project, error) {
 
 	sql :=
 		`select distinct p.project_id, p.name, p.public, 
-			p.owner_id, p.creation_time, p.update_time, pm.role role
```
Comment (reasonerjt) [inline src/common/dao/project.go]: no impact for other UI/API?

File: src/ui/api/search.go [hunk 2]
```diff
@@ -77,26 +77,36 @@ func (s *SearchAPI) Get() {
...
 		}
-		if match {
-			if userID != dao.NonExistUserID {
-				if isSysAdmin {
-					p.Role = models.PROJECTADMIN
-				}
-				if p.Role == models.PROJECTADMIN {
-					p.Togglable = true
-				}
+		if !match {
```
Comment (reasonerjt) [inline src/ui/api/search.go]: that variable "match" is not needed

File: src/common/dao/project.go (hunk 1 above)
Comment (reasonerjt) [inline src/common/dao/project.go]: Will it impact other API?

File: src/ui/api/search.go (hunk 2 above)
Comment (reasonerjt) [inline src/ui/api/search.go]: Seems the variable "match" is not needed.

File: src/ui/api/search.go [hunk 3]
```diff
@@ -77,26 +77,36 @@ func (s *SearchAPI) Get() {
...
+			continue
+		}
+
+		if userID != dao.NonExistUserID {
+			if isSysAdmin {
+				p.Role = models.PROJECTADMIN
 			}
 
-			repos, err := dao.GetRepositoryByProjectName(p.Name)
+			roles, err := dao.GetUserProjectRoles(userID, p.ProjectID)
```
Comment (reasonerjt) [inline src/ui/api/search.go]: if isSysAdmin == True this is not needed?

File: src/common/dao/project.go (hunk 1 above)
Comment (ywk253100) [inline src/common/dao/project.go]: No

**Comment by reasonerjt** [comment]
Please add more test case to test the search API.

**Pull request merged**